monkey/monkey_island/cc/server_setup.py

```diff
@@ -88,6 +93,18 @@ def _initialize_globals(config_options: IslandConfigOptions, server_config_path:
     initialize_services(config_options.data_dir)
 
 
+def _configure_gevent_exception_handling(logger, data_dir):
+    hub = gevent.hub.get_hub()
+
+    gevent_exception_log = open(data_dir / GEVENT_EXCEPTION_LOG, "a+")
+    atexit.register(gevent_exception_log.close)
+
+    # Send gevent's exception output to a log file.
+    # https://www.gevent.org/api/gevent.hub.html#gevent.hub.Hub.exception_stream
+    hub.exception_stream = gevent_exception_log
+    hub.handle_error = GeventHubErrorHandler(hub, logger)
```

Comment:
Why pass `logger` as an argument here? Why not have `logger = logging.getLogger(__name__)` in `monkey_island/cc/setup/gevent_hub_error_handler.py`?

Reply:
So that all of the gevent objects (i.e. Hub and WSGIServer) have the same logger. It also has the added benefit of making the dependencies of `GeventHubErrorHandler` explicit. Unlike other classes, this class doesn't use a logger to log information about itself. Rather, it uses a logger object as part of an exception handling routine on behalf of another component.